pkg/client/kubeconfig_test.go

```diff
@@ -0,0 +1,173 @@
+package client
+
+import (
+	"os"
+
+	mocks "github.com/Azure/aks-tls-bootstrap-client/pkg/client/mocks"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"go.uber.org/mock/gomock"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+const (
+	dummyCertPem = `-----BEGIN CERTIFICATE-----
```

Comment:
we'll want coverage for valid certs and expired certs, etc. - might be better to just generate the certs at test-time instead of hardcoding her